piptools/writer.py

```diff
@@ -258,15 +277,15 @@ def _format_requirement(
             required_by.add(_comes_from_as_string(ireq))
 
         if required_by:
-            sorted_required_by = sorted(required_by)
-            if len(sorted_required_by) == 1:
-                source = sorted_required_by[0]
-                annotation = "    # via " + source
+            if self.annotation_style == "line":
```

Comment:
I'd prefer this instead. Works fine with coverage and explicitly checks all annotation styles (even erroneous):

```python
            if self.annotation_style == "line":
                annotation = annotation_style_line(required_by)
                sep = "\n    " if ireq_hashes else "  "
            elif self.annotation_style == "split":
                annotation = annotation_style_split(required_by)
                sep = "\n    "
            else:  # pragma: no cover
                raise ValueError('Invalid value for annotation style')
```

Reply:
As an alternative option, we could add a unit test to `test_writer.py` to check `else`  branch instead of using `no cover` pragma. Up to you. Both are fine by me.

Reply:
Excellent, thanks, let's go with the `no cover` approach.